python/cudf/cudf/utils/ioutils.py

```diff
@@ -2099,3 +2108,96 @@ def _read_byte_ranges(
 
     for worker in workers:
         worker.join()
+
+
+def _get_remote_bytes_all(
+    remote_paths, fs, *, blocksize=_BYTES_PER_THREAD_DEFAULT
+):
+    # TODO: Experiment with a heuristic to avoid the fs.sizes
+    # call when we are reading many files at once (the latency
+    # of collecting the file sizes is unnecessary in this case)
+    if max(sizes := fs.sizes(remote_paths)) <= blocksize:
+        # Don't bother breaking up individual files
+        return fs.cat_ranges(remote_paths, None, None)
+    else:
+        # Construct list of paths, starts, and ends
+        paths, starts, ends = zip(
+            *[
+                (r, j, min(j + blocksize, s))
+                for r, s in zip(remote_paths, sizes)
+                for j in range(0, s, blocksize)
+            ]
+        )
+
+        # Collect the byte ranges
+        chunks = fs.cat_ranges(paths, starts, ends)
+
+        # Construct local byte buffers
+        buffers = [
+            functools.reduce(operator.add, chunks[:counts])
+            for _, counts in zip(
+                remote_paths, np.unique(paths, return_counts=True)[1]
+            )
```

Comment:
*issue (potentially)*: The return value of `np.unique` is in the _sorted_ order of the paths. If any of `paths` are _not_ in order, then `chunks` might not line up with `counts`.

Also, minor:

```
[x for _, x in zip(ignored, foo)]
```
Is better written as:
```
foo
```

And so:
```
buffers = [
    functools.reduce(operator.add, chunks[:count])
    for count in np.unique(paths, return_counts=True)[1]
]
```

Reply:
Thanks for catching this! It looks like `functools.reduce(operator.add, chunks[:counts])` is also a bit "wrong" - I'll try to revise this logic.

Reply:
Whoops sorry about that.